presto-main/src/main/java/com/facebook/presto/operator/aggregation/partial/SkipAggregationBuilder.java

```diff
@@ -0,0 +1,183 @@
+    }
+
+    @Override
+    public boolean isFull()
+    {
+        return currentPage != null;
+    }
+
+    @Override
+    public void updateMemory()
+    {
+        if (currentPage != null) {
+            memoryContext.setBytes(currentPage.getSizeInBytes());
+        }
+    }
+
+    @Override
+    public void recordHashCollisions(HashCollisionsCounter hashCollisionsCounter)
+    {
+        // no op
+    }
+
+    @Override
+    public void close()
+    {
+    }
+
+    @Override
+    public ListenableFuture<?> startMemoryRevoke()
+    {
+        throw new UnsupportedOperationException("startMemoryRevoke not supported for SkipAggregationBuilder");
+    }
+
+    @Override
+    public void finishMemoryRevoke()
+    {
+        throw new UnsupportedOperationException("finishMemoryRevoke not supported for SkipAggregationBuilder");
+    }
+
+    private Page buildOutputPage(Page page)
+    {
+        populateInitialAccumulatorState(page);
+        BlockBuilder[] outputBuilders = serializeAccumulatorState(page.getPositionCount());
+        return constructOutputPage(page, outputBuilders);
+    }
+
+    private void populateInitialAccumulatorState(Page page)
+    {
+        int positionCount = page.getPositionCount();
+        GroupByIdBlock groupByIdBlock = new GroupByIdBlock(positionCount, new LongArrayBlock(positionCount, Optional.empty(), consecutive(positionCount)));
+        for (GroupedAccumulator groupedAccumulator : groupedAccumulators) {
+            groupedAccumulator.addInput(groupByIdBlock, page);
+        }
+    }
+
+    private BlockBuilder[] serializeAccumulatorState(int positionCount)
+    {
+        BlockBuilder[] outputBuilders = new BlockBuilder[groupedAccumulators.size()];
+        for (int i = 0; i < outputBuilders.length; i++) {
+            outputBuilders[i] = groupedAccumulators.get(i).getIntermediateType().createBlockBuilder(null, positionCount);
+        }
+
+        for (int position = 0; position < positionCount; position++) {
+            for (int i = 0; i < groupedAccumulators.size(); i++) {
+                GroupedAccumulator groupedAccumulator = groupedAccumulators.get(i);
+                BlockBuilder output = outputBuilders[i];
+                groupedAccumulator.evaluateIntermediate(position, output);
+            }
+        }
+
+        return outputBuilders;
+    }
+
+    private Page constructOutputPage(Page page, BlockBuilder[] outputBuilders)
+    {
+        Block[] outputBlocks = new Block[hashChannels.length + outputBuilders.length];
+        for (int i = 0; i < hashChannels.length; i++) {
+            outputBlocks[i] = page.getBlock(hashChannels[i]);
+        }
+        for (int i = 0; i < outputBuilders.length; i++) {
+            outputBlocks[hashChannels.length + i] = outputBuilders[i].build();
+        }
+        return new Page(page.getPositionCount(), outputBlocks);
+    }
+
+    private static long[] consecutive(int positionCount)
+    {
+        long[] longs = new long[positionCount];
```

Comment:
fillConsecutive ? or Range to borrow the python name ?

Reply:
renamed to fillConsecutive